service/history/workflow/context.go

```diff
@@ -680,7 +680,7 @@ func (c *ContextImpl) SetWorkflowExecution(ctx context.Context, now time.Time) (
 		return err
 	}
 	if len(resetWorkflowEventsSeq) != 0 {
-		return serviceerror.NewInternal("SetWorkflowExecution encountered new events")
+		c.logger.Warn("SetWorkflowExecution encountered new events")
```

Comment:
nit: also emit a metrics?

and we should revert the change before next release (1.19)

Reply:
Added a metrics